src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, Input, OnInit, SimpleChanges } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { Attribute } from '../../entities/attribute.type';
+import { DownloadedModel } from '../../entities/downloaded-model.type';
+import { Embedding } from '../../entities/embedding.type';
+import { DataHandlerHelper } from '../../helper/data-handler-helper';
+import { SettingModals } from '../../helper/modal-helper';
+
+@Component({
+  selector: 'kern-embeddings',
+  templateUrl: './embeddings.component.html',
+  styleUrls: ['./embeddings.component.scss'],
+})
+export class EmbeddingsComponent implements OnInit {
+
+  @Input() project: Project;
+  @Input() attributesArrayTextUsableUploaded: Attribute[] = [];
+  @Input() settingModals: SettingModals;
+  @Input() isManaged: boolean;
+  @Input() embeddings: Embedding[] = [];
+  @Input() embeddingHandlesMap: Map<string, any> = new Map<string, any>();
+
+  downloadedModelsList$: any;
+  downloadedModelsQuery$: any;
+  downloadedModels: DownloadedModel[] = [];
+  subscriptions$: Subscription[] = [];
+  dataHandlerHelper: DataHandlerHelper;
```

Comment:
all of these seem a bit fishy:
- downloadedModelsList$: any; -> value changed -> only used once to init subscription local variable would be enough
- downloadedModelsQuery$: any; -> query -> no websocket stuff so never used again
- downloadedModels: DownloadedModel[] = []; initialized with an empty array though reset on every subscription call -> checkIfModelIsDownloaded also checks for undefined so shouldn't be needed to initialze
- subscriptions$: Subscription[] = []; -> are set but never unsubscribed
- dataHandlerHelper: DataHandlerHelper; -> maybe as input if absolutely necessary but not created new (since this will run the constructor again but only the constructor so the necessary variables aren't filled)